### ISSUE TITLE
Remove monthly_supporter from user endpoints except users/me

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistAPI#479]
Currently we're including the monthly_supporter attribute at https://api.inaturalist.org/v2/users/1?fields=all (and maybe at other endpoints?). That describes whether the user *is* a monthly supporter, but we also have an opt-in preference to control whether this information is displayed, implying that it's at least somewhat secret. So I suggest we

- [ ] Remove `monthly_supporter` from all endpoints other than users/me
- [ ] Add `prefers_monthly_supporter_badge` to GET /users/:id

**Comment by kvangork** [comment]
@kueda Will our existing apps cleanly handle `monthly_supporter` disappearing from responses?
If not, or if we're unsure how other code will react, we could leave it in place but change the response value so that it returns the true status if `:id == me` but otherwise respects the logic you proposed for the `monthly_supporter_badge`

**Comment by kueda** [comment]
@budowski and @alexshepard will have a more informed opinion, but neither of the legacy app repos make direct reference to `monthly_supporter`.

**Comment by budowski** [comment]
Yeah, like @kueda said, Android doesn't make use of that property - we could leave that property on for id=me like @kvangork suggested, if we plan on displaying that badge inside the app somewhere.

**Comment by alexshepard** [comment]
We don't map this from json, so the legacy iOS app wont notice if it goes away.
